### ISSUE TITLE
Remove deep copy in recursive function

### ISSUE THREAD

**Issue by thewahome** [microsoftgraph/microsoft-graph-explorer-v4#2152]
**Describe the bug**
[microsoft-graph-explorer-v4/resources-filter.ts at dev · microsoftgraph/microsoft-graph-explorer-v4 (github.com)](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/blob/dev/src/app/utils/resources/resources-filter.ts#L9)
 
This call is deserializing/serializing trees and subtrees in a recursive fashion. It's expensive, memory consuming, and basically useless. It can be done once before getting in that function and the work is done. This is invoked every time you change a character that is not part of the query string. Since an object is a tree, the dump/parse loop is basically done on sub-items in a repetitive fashion.

**Comment by thewahome** [comment]
As pointed out by @MaximRouiller
